src/averageAges.js

```diff
@@ -1,61 +1,40 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const personCenture = people.filter(person => century
+    ? Math.ceil(person.died / 100) === century && person.sex === 'm'
+    : person.sex === 'm');
+
+  const averageAgeMen = personCenture.reduce((sum, years) =>
+    sum + (years.died - years.born), 0) / personCenture.length;
+
+  return averageAgeMen;
 }
+// _____________________________________________________________
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const personWomen = people.filter(person => withChildren
+    ? people.some(child => child.mother === person.name) && person.sex === 'f'
+    : person.sex === 'f');
+
+  const averageAgeWomen = personWomen.reduce((sum, years) =>
+    sum + (years.died - years.born), 0) / personWomen.length;
+
+  return averageAgeWomen;
 }
+// ___________________________________________________________
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(person => onlyWithSon
+    ? people.some(mother => person.mother === mother.name) && person.sex === 'm'
+    : people.some(mother => person.mother === mother.name)
+  );
```

Comment:
Looks good👍

Reply:
no, looks like you duplicate yourself, don't filter by parent and sex at a time, split into several tasks